src/libraries/Microsoft.PowerFx.Core/Texl/Remove.cs

```diff
@@ -0,0 +1,451 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Entities;
+using Microsoft.PowerFx.Core.Entities.QueryOptions;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Functions.DLP;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Logging.Trackers;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Types.Enums;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    internal abstract class RemoveBaseFunction : BuiltinFunction
+    {
+        public override bool ManipulatesCollections => true;
+
+        public override bool ModifiesValues => true;
+
+        public override bool CanSuggestInputColumns => true;
+
+        public override bool IsSelfContained => false;
+
+        public override bool RequiresDataSourceScope => true;
+
+        public override bool SupportsParamCoercion => false;
+
+        public override RequiredDataSourcePermissions FunctionPermission => RequiredDataSourcePermissions.Delete;
+
+        public override bool MutatesArg(int argIndex, TexlNode arg) => argIndex == 0;
+
+        public override bool TryGetTypeForArgSuggestionAt(int argIndex, out DType type)
+        {
+            if (argIndex > 0)
+            {
+                type = default;
+                return false;
+            }
+
+            return base.TryGetTypeForArgSuggestionAt(argIndex, out type);
+        }
+
+        public RemoveBaseFunction(int arityMax, params DType[] paramTypes)
```

Comment:
[minor] Can you move those function declarations after the constructor for consistency?